src/controllers/customer.ts

```diff
@@ -106,4 +119,78 @@ export class AccountController {
+        // 3. Check the token balance for Testnet account
+        // 3.1 If it's less then required for DID creation - assign new portion from testnet-faucet
+        const customerId: string = response.locals.customerId || LogToWebHook.getCustomerId(request);
+        const logToHelper = new LogToHelper()
+        const _r = await logToHelper.setup()
+        if (_r.status !== StatusCodes.OK) {
+            return response.status(StatusCodes.BAD_GATEWAY).json({
+                error: _r.error})
+        }
+        // 1. Check that the customer exists
+        let customer : any = await CustomerService.instance.get(customerId)
+        if (!customer) {
+            customer = await CustomerService.instance.create(customerId)
+        }
+        // 2. Get the user's roles
+        const roles = await logToHelper.getRolesForUser(customerId)
+        if (roles.status !== StatusCodes.OK) {
+            return response.status(StatusCodes.BAD_GATEWAY).json({
+                error: roles.error})
+        }
+
+        // 2.1 If list of roles is empty and the user is not suspended - assign default role
+        if (roles.data.length === 0 && !LogToWebHook.isUserSuspended(request)) {
+            const _r = await logToHelper.setDefaultRoleForUser(customerId)
+            if (_r.status !== StatusCodes.OK) {
+                return response.status(StatusCodes.BAD_GATEWAY).json({
+                    error: _r.error})
+            }
+        }
+
+        const customDataFromLogTo = await logToHelper.getCustomData(customerId)
+
+        // 2.2 Create custom_data and update the userInfo (send it to the LogTo)
+        if (Object.keys(customDataFromLogTo.data).length === 0 && customer.address) {
+            const customData = {
+                cosmosAccounts: {
+                    testnet: customer.address
+                }
+            }
+            const _r = await logToHelper.updateCustomData(customerId, customData)
+            if (_r.status !== 200) {
+                return response.status(_r.status).json({
+                    error: _r.error})
+            }
+        }
+
+        // 3. Check the token balance for Testnet account
+        if (customer.address && process.env.FAUCET_ENABLED === 'true') {
+            const balance = await AccountController.checkBalance(customer.address)
+            if (!balance || +balance.amount < process.env.TESTNET_MINIMUM_BALANCE) {
+                // 3.1 If it's less then required for DID creation - assign new portion from testnet-faucet
+                const resp = await FaucetHelper.delegateTokens(customer.address)
+                if (resp.status !== StatusCodes.OK) {
+                    return response.status(StatusCodes.BAD_GATEWAY).json({
+                        error: resp.error})
+                }
+            }
+        }
+        return response.status(StatusCodes.OK).json({})
+    }
+
+    public static async checkBalance(address: string, network = NetworkType.Testnet) {
```

Comment:
@anikitinDSR It's better to move this as a util function to our sdk. Even the getCosmosAccount. As these functions can be resused and can be managed easily when there are breaking changes in cosmjs